snow/engine/snowman/event/queue_test.go

```diff
@@ -0,0 +1,294 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package event
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+var (
+	errDuplicateInvocation  = errors.New("job already handled")
+	errUnexpectedInvocation = errors.New("job handled unexpectedly")
+)
+
+type testJob struct {
+	execute func(context.Context) error
+	cancel  func(context.Context) error
+}
+
+func (j *testJob) Execute(ctx context.Context) error {
+	return j.execute(ctx)
+}
+
+func (j *testJob) Cancel(ctx context.Context) error {
+	return j.cancel(ctx)
+}
+
+func newQueueWithJob[T comparable](t *testing.T, job Job, shouldCancel bool, dependencies ...T) *Queue[T] {
+	q := NewQueue[T]()
+	require.NoError(t, q.Register(context.Background(), job, dependencies...))
+	if shouldCancel {
+		for _, jobs := range q.jobs {
+			for _, j := range jobs {
+				j.shouldCancel = true
+			}
+		}
+	}
+	return q
+}
+
+func TestQueue_Register(t *testing.T) {
+	var calledExecute bool
+	userJob := &testJob{
+		execute: func(context.Context) error {
+			if calledExecute {
+				return errDuplicateInvocation
+			}
+			calledExecute = true
+			return nil
+		},
+		cancel: func(context.Context) error {
+			return errUnexpectedInvocation
+		},
+	}
+
+	tests := []struct {
+		name          string
+		queue         *Queue[int]
+		dependencies  []int
+		shouldExecute bool
```

Comment:
(style,nit) `wantExecute`; "should" implies that it's a parameter for the test loop whereas "want" is in keeping with idiomatic `got` and `want` (in place of `actual` and `expected`).

If it were me I'd also changed `expected*` to `want*` but I don't think it's as important since `expected` still conveys intent unambiguously.

Reply:
Reading through the tests I now see that it's always false. I think you can remove it entirely, but my original comment re the prefix applies to the other tests so I'll leave it.

Reply:
`shouldExecute` is set to `true` in the first case

Reply:
I agree with marun re consistency of `expect` over `want`.

> shouldExecute is set to true in the first case

Whoops!